RouteClassError is called when req.get_arg fails

If there are no arguments in the URI path & `req.get_arg` is called then `RouteClassError` is raised. 

Fix:
provide a new custom exception for this type of error.

> hey @joegasewicz, can you provide a sample test case?
> 
> I was trying with something like this but not getting RouteClassError
> 
> ```python
> def test_get_args_with_no_uri_path_args(self, bobtail_app, environ):
>         class Images:
> 
>             def get(self, req, res):
>                 pass
> 
>         routes = [
>             (Images(), "/images")
>         ]
> 
>         app = bobtail_app(routes=routes)
> 
>         env = environ(path="/images")
>         data = app(env, lambda s, r: None)
>         id = app.request.get_arg("id")
>         assert id == 1
> ```

If i have the following route:

```python
def get(self, req: Request, res: Response):
        req.get_arg("apples")
```

The registered route for this handler is `http://127.0.0.1:8000/images/{fruit:str}`  - obviously, for this handler to work i need to create a request like this -  `http://127.0.0.1:8000/images/apples`  

But if i just make a request using `http://127.0.0.1:8000/images` without the request arg then the `bobtail.exceptions.RouteClassError` is raised.

This is the stack trace:

```
[2022-12-03 11:32:53 +0000] [7307] [ERROR] Error handling request /images
Traceback (most recent call last):
  File "/home/joe/.local/share/virtualenvs/file_server-a96Bzucb/lib/python3.11/site-packages/bobtail/wsgi.py", line 63, in _call_handler
    self.middleware.call(self.request, self.response, handler)
  File "/home/joe/.local/share/virtualenvs/file_server-a96Bzucb/lib/python3.11/site-packages/bobtail/middleware.py", line 40, in call
    self.handler(self.request, self.response)
  File "/home/joe/GolandProjects/nottoboard/file_server/api/routes/images.py", line 8, in get
    req.get_arg("apples")
  File "/home/joe/.local/share/virtualenvs/file_server-a96Bzucb/lib/python3.11/site-packages/bobtail/request.py", line 40, in get_arg
    arg_value = self.args[name]["value"]
                ~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/joe/.local/share/virtualenvs/file_server-a96Bzucb/lib/python3.11/site-packages/gunicorn/workers/sync.py", line 136, in handle
    self.handle_request(listener, req, client, addr)
  File "/home/joe/.local/share/virtualenvs/file_server-a96Bzucb/lib/python3.11/site-packages/gunicorn/workers/sync.py", line 179, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joe/.local/share/virtualenvs/file_server-a96Bzucb/lib/python3.11/site-packages/bobtail/wsgi.py", line 103, in __call__
    self._handle_route()
  File "/home/joe/.local/share/virtualenvs/file_server-a96Bzucb/lib/python3.11/site-packages/bobtail/wsgi.py", line 80, in _handle_route
    self._call_handler(route, "get")
  File "/home/joe/.local/share/virtualenvs/file_server-a96Bzucb/lib/python3.11/site-packages/bobtail/wsgi.py", line 65, in _call_handler
    raise RouteClassError("route class is not instantiated") from exc
bobtail.exceptions.RouteClassError: route class is not instantiated
```

--- a/bobtail/request.py
+++ b/bobtail/request.py
@@ -25,6 +25,7 @@ class Request(ABC):
         self.path = path
         self.method = method
         self.headers = headers
+        self.args: Dict = {}
         self.wsgi_input = WSGIInput(
             byte_data=byte_data,
             headers=self.headers,
@@ -37,6 +38,9 @@ class Request(ABC):
         :return:
         :rtype:
         """
+        if name not in self.args:
+            return None
+
         arg_value = self.args[name]["value"]
         match self.args[name]["type"]:
             case "int":

--- a/tests/test_request.py
+++ b/tests/test_request.py
@@ -28,6 +28,27 @@ class TestRequest:
         assert name == "hello"
         assert is_raining is True
 
+    def test_get_args_if_key_not_present(self, bobtail_app, environ):
+        class Images:
+
+            def get(self, req, res):
+                pass
+
+        routes = [
+            (Images(), "/images/{id:int}")
+        ]
+
+        app = bobtail_app(routes=routes)
+
+        env = environ(path="/images/1")
+        data = app(env, lambda s, r: None)
+        assert app.request.get_arg("id") == 1
+        assert app.request.get_arg("name") == None
+
+        env = environ(path="/images")
+        data = app(env, lambda s, r: None)
+        assert app.request.get_arg("id") == None
+
     def test_get_json(self):
         req_headers = RequestHeaders("application/json")
         req = Request(
@@ -97,5 +118,3 @@ class TestRequest:
             },
         }
         assert result == expected
-
-

